src/scripts/main.js

```diff
@@ -1,3 +1,54 @@
+/* eslint-disable no-unused-expressions */
 'use strict';
 
-// write code here
+const tableHead = document.querySelector('thead');
+const tableBody = document.querySelector('tbody');
+
+tableHead.addEventListener('click', (e) => {
+  const columnName = e.target.closest('th');
+  const rows = tableBody.rows;
+  const people = [];
+
+  for (const row of rows) {
+    people.push({
+      Name: row.cells[0].textContent,
+      Position: row.cells[1].textContent,
+      Age: row.cells[2].textContent,
+      Salary: row.cells[3].textContent,
+    });
+  }
+
+  const sortedName = columnName.textContent.trim();
+
+  people.sort((person1, person2) => {
+    if (sortedName === 'Name' || sortedName === 'Position') {
+      return person1[sortedName].localeCompare(person2[sortedName]);
+    }
+
+    if (columnName.textContent === 'Age') {
+      return parseInt(person1[sortedName]) - parseInt(person2[sortedName]);
+    }
```

Comment:
The condition `if (columnName.textContent === 'Age')` should be `if (sortedName === 'Age')` to maintain consistency with the other conditions and ensure the correct column is being checked.